### ISSUE TITLE
Old k8s dependency version?

### ISSUE THREAD

**Issue by ipleten** [vmware-tanzu/octant#2666]
Isn't 0.22 version supposed to have newer k8s library?

It's v1.19.3 
https://github.com/vmware-tanzu/octant/blame/v0.22.0/go.mod#L58
While 1.21.3 stable is already available.

**What did you expect to happen:**
I suspect https://github.com/kubernetes/kubernetes/pull/97857 released in 1.21 might help to solve 

> W0719 15:24:46.802086   74811 exec.go:271] constructing many client instances from the same exec auth config can cause performance problems during cert rotation and can exhaust available network connections; 1229 clients constructed calling "aws-iam-authenticator"
W0719 15:24:46.824047   74811 exec.go:271] constructing many client instances from the same exec auth config can cause performance problems during cert rotation and can exhaust available network connections; 1230 clients constructed calling "aws-iam-authenticator"
W0719 15:24:46.872156   74811 exec.go:271] constructing many client instances from the same exec auth config can cause performance problems during cert rotation and can exhaust available network connections; 1231 clients constructed calling "aws-iam-authenticator"
W0719 15:24:47.053590   74811 exec.go:271] constructing many client instances from the same exec auth config can cause performance problems during cert rotation and can exhaust available network connections; 1232 clients constructed calling "aws-iam-authenticator"

- Octant version (use `octant version`): 
Version:  0.22.0
Git commit:  a723e6e682f8d73d9c2afdd033c9028c9610b025
Built:  2021-07-14T12:20:03Z

**Comment by GuessWhoSamFoo** [comment]
Duplicate of #1558

**Comment by ipleten** [comment]
Is there any roadmap or plan on updating that?
I can send PR bumping these libs but not sure about versioning and which version of k8s is supported by each octant version. Documentation isn't very clear about that:

> Octant follows an n±1 policy for versions of Kubernetes similar to kubectl. For example, Octant 0.16.0 uses the Kubernetes 1.19 client. So version 0.16.0 can be used with Kubernetes 1.18, 1.19, and 1.20.

Does it mean that octant supports 3 last minor versions of k8s? If yes, does it imply that it should be 3 active octant release branches?

**Comment by davidmirror-ops** [comment]
What we discussed during today's community meeting, where we brought up this question, goes like this:

1. Take the client-go version of the Octant release (1.19 for Octant 0.22)
2. That particular Octant release will support Kubernetes versions in a n±1 model, in this case it will be:

1.18 --  1.19 --- 1.20

We agree it could be better phrased in the docs, and we will be improving in that area in the upcoming weeks and months.

Octant is aimed to be n-1 compared with Kubernetes upstream versions, but also if there's no blocking issue, an upgrade on that versioning could be evaluated.

@ipleten please let us know if more details are required.

**Comment by davidmirror-ops** [comment]
Also adding great comment from @vrabbi on the Slack channel:

Regarding client go version upgrades:
The one worry could be regarding managed clusters. For example eks just changed the default to 1.20 and 1.21 isnt even available yet as far as i know. Because managed clusters are always lagging behind in k8s versions from the upstream versions, it could make it a bit of a pain for those users. They could still use an old version of octant but then they wouldnt get the bug fixes and new features. Also as more and more plugins exist, if the plugin apis have breaking changes it could make it difficult to utilize plugins in such use cases

**Comment by GuessWhoSamFoo** [comment]
tldr; PR for a bump to 1.21.3 is appreciated

**Comment by ipleten** [comment]
https://github.com/kubernetes/client-go/issues/914

**Comment by wwitzel3** [comment]
Thank you for the PR.
